openssl/aes.go

```diff
@@ -116,32 +107,19 @@ func (c *aesCipher) Decrypt(dst, src []byte) {
 		panic("crypto/aes: output not full block")
 	}
 	if c.dec_ctx == nil {
-		c.dec_ctx = C._goboringcrypto_EVP_CIPHER_CTX_new()
-		if c.dec_ctx == nil {
-			panic("cipher: unable to create EVP cipher ctx")
-		}
-
-		k := (*C.uchar)(unsafe.Pointer(&c.key[0]))
-
-		if C.int(1) != C._goboringcrypto_EVP_CipherInit_ex(c.dec_ctx, c.cipher, nil, k, nil, C.GO_AES_DECRYPT) {
-			panic("cipher: unable to initialize EVP cipher ctx")
+		var err error
+		c.dec_ctx, err = newCipherCtx(c.cipher, C.GO_AES_DECRYPT, c.key, nil)
+		if err != nil {
+			panic(err)
 		}
 	}
-	// Workaround - padding detection is broken but we don't need it
```

Comment:
I haven't found the need to implement this workaround. Could you give more context or share a failing test case?

Reply:
The original catalyst for this change if I'm not mistaken is due to: https://bugzilla.redhat.com/show_bug.cgi?id=1979100. A lot of the discussion on that thread is private but that is what led to the change.

Reply:
Thanks for the context. I can't reproduce the issue with the current implementation, but I'm fine adding the workaround just in case. Anyway, it makes sense to explicitly disable padding if Go expects full blocks.

Just one comment: is there any reason `EVP_CIPHER_CTX_set_padding` is not inside the previous block, which is just called once to initialize the context?

Reply:
I don't believe so, it should be able to be hoisted.

Reply:
Done.